sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -266,27 +292,135 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                                  List<Map<String, String>> sortDocument,
                                  Map<String, Object> indexes,
                                  SQLDatabase db) {
-        // TODO - sort logic...
-        return null;
+        boolean smallResultSet = (docIdSet.size() < SMALL_RESULT_SET_SIZE_THRESHOLD);
+        SqlParts orderBy = sqlToSortIds(docIdSet, sortDocument, indexes);
+        List<String> sortedIds = new ArrayList<String>();
```

Comment:
Probably move the assignment inside the `if`, so we can return `null` without the `else`, or mirror the assignment to `nil` in the objc code rather than doing an early return.